src/Microsoft.Health.Fhir.TemplateManagement/Utilities/OrasUtility.cs

```diff
@@ -0,0 +1,98 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Health.Fhir.TemplateManagement.Exceptions;
+using Microsoft.Health.Fhir.TemplateManagement.Models;
+
+namespace Microsoft.Health.Fhir.TemplateManagement.Utilities
+{
+    public static class OrasUtility
+    {
+        public static string GetOrasFileName()
+        {
+            string orasFileName;
+            switch (Environment.OSVersion.Platform)
+            {
+                case PlatformID.Win32NT:
+                    orasFileName = "oras-win.exe";
+                    break;
+                case PlatformID.Unix:
+                    orasFileName = "oras-unix";
+                    AddFilePermissionInLinuxSystem(orasFileName);
+                    break;
+                default:
+                    throw new TemplateManagementException("Operation system is not supported");
+            }
+
+            return orasFileName;
+        }
+
+        public static async Task OrasExecutionAsync(string command, string orasFileName, string orasWorkingDirectory)
+        {
+            TaskCompletionSource<bool> eventHandled = new TaskCompletionSource<bool>();
```

Comment:
Can we remove the GetOrasFileName function and just check the platform information during each execution? The caller should not care about whether windows or unix binary is used.

Reply:
Seems we don't need this utility? We can put this logic in the OrasClient